api-connection.html

```diff
@@ -100,7 +100,203 @@ <h2 id="function_newConnection">new Connection(config)</h2>
       <dd>
         Type of the authentication method, valid types are <code>default</code>, <code>ntlm</code>,
         <code>azure-active-directory-password</code>, <code>azure-active-directory-access-token</code>, 
-        <code>azure-active-directory-msi-vm</code>, or <code>azure-active-directory-msi-app-service</code> 
+        <code>azure-active-directory-msi-vm</code>, <code>azure-active-directory-service-principal-secret</code>,
+        or <code>azure-active-directory-msi-app-service</code> 
+        <ul>
+          <li><code>azure-active-directory-password</code></li>
+{% highlight javascript %}
+"authentication": {
+  ...,
+  "options": {
+    "userName": value;
+    "password": value;
+    "domain": value; (Optional)
+  }
+}           
+{% endhighlight %}
+          <ul>
+            <li>username (string): A user need to provide `userName` associated to their account.</li>
+            <li>password (string): A user need to provide `password` associated to their account.</li>
+            <li>domain (string): Optional parameter for specific Azure tenant ID.</li>
+          </ul>
+          <li><code>azure-active-directory-access-token</code></li>
+{% highlight javascript %}
+"authentication": {
+  ...,
+  "options": {
+    "token": value;
+  }
+}         
+{% endhighlight %}
+          <ul>
+            <li>token (string): A user need to provide `token` which they retrieved elsewhere to forming the connection.</li>
+            <p><h4>How to get the token</h4></p>
+            <ul>
+              <h4>Things that need to considered before using the fallowing scripts:</h4>
+              <ul>
+                <ol>
+                    <li>
+                      Two PowerShell scripts need to be run within a msi-vm or a msi-app-service environment,since it need to pick up the specific preset environment variables from the system or reach a specific msi end point.
```

Comment:
```suggestion
                      Two PowerShell scripts need to be run within an msi-vm or an msi-app-service environment, since it needs to pick up the specific preset environment variables from the system or reach a specific msi endpoint.
```